### DISCUSSION TITLE
How can I implement a context menu when right-clicking a table row?

### DISCUSSION THREAD

**Discussion by BadaHanDL** [cloudscape-design/components#1709] [Q&A] [answered]
Using the variant="icon" prop in the ButtonDropdown component, I can display the actions menu when a row is clicked. Now, how can I display the same menu as a context menu when right-clicking on a row?

Thank you,
Bada Han

**Comment by just-boris** [comment, marked as answer]
Custom dropdown menus is not a standard UI feature on web interfaces. I would recommend you to avoid custom right-click menus, because it is not only misaligned with our design system, but with web design in general

**Reply by BadaHanDL** [reply]
Could you please let me know how to expand the menu by default on the right-click of an element?

Currently, the three dots vertical button is displayed for each row of a table, and clicking it expands the groups of actions. But, my requirement is to expand the group of actions (menu) directly on the right click.

Code:
import * as React from "react";
import ButtonDropdown from "@cloudscape-design/components/button-dropdown";

export default () => {
  return (
    <ButtonDropdown
      items={[
        { id: "start", text: "Start" },
        { id: "stop", text: "Stop", disabled: true },
        {
          id: "hibernate",
          text: "Hibernate",
          disabled: true
        },
        { id: "reboot", text: "Reboot", disabled: true },
        { id: "terminate", text: "Terminate" }
      ]}
      ariaLabel="Control instance"
      variant="icon"
    />
  );
}

**Reply by just-boris** [reply]
We do not support right click overrides for the reasons I posted above. I cannot give a guidance on how to do unsupported stuff
